### PR TITLE
fix issue_2588

### PR DESCRIPTION
**Pull request by AymenFJA**
Related to #2588.

### CODE REVIEW

File: docs/source/machconf.rst [hunk 1]
```diff
@@ -195,6 +196,14 @@ All fields are mandatory, unless indicated otherwise below.
 * ``pre_bootstrap_0``: list of commands to execute for initialization of main agent (optional).
 * ``pre_bootstrap_1``: list of commands to execute for initialization of sub-agent (optional).
 * ``forward_tunnel_endpoint``: name of the host which can be used to create ssh tunnels from the compute nodes to the outside world (optional).
+* ``virtenv_mode``: RADICAL-Pilot agent enviroment setup (optional):
+        * ``create``: create a python virtual enviroment from scratch (default).
+        * ``recreate``: delete the exsiting virtual enviroment and build it from scratch, if not found then ``create``.
+        * ``use``: use an existing virtual enviroment, if not found then ``create``.
+        * ``update`` update the existing virtual enviroment, if not found then ``create``.
+        * ``local``: use the client existing virtual enviroment.
+* ``virtenv``: only use when ``virtenv_mode=use``, path to an exsiting python enviroment with pre-installed RCT stack (optional).
```
Comment (eirrgang) [inline docs/source/machconf.rst:205]: Is `virtenv_dist` still a thing?

**Review by eirrgang — COMMENTED** [review]
There are still undocumented fields, such as "cores_per_node". Has the resource configuration data structure been formalized, yet, as a class that could be documented instead?

File: docs/source/machconf.rst (hunk 1 above)
Comment (andre-merzky) [reply docs/source/machconf.rst:205]: Alas it will remain a thing for a while...

**Review by andre-merzky — APPROVED** [review]
Thanks @eirrgang

File: docs/source/machconf.rst (hunk 1 above)
Comment (mturilli) [reply docs/source/machconf.rst]: @andre-merzky ping

Comment (andre-merzky) [reply docs/source/machconf.rst]: `By default, RP will install virtualenv version '1.9' while bootstrapping the pilot.  setting 'virtenv_dist` to a different version string will use that respective version.  Setting it to the string 'system' will use the system supplied 'virtualenv' command.`

**Comment by andre-merzky** [comment]
Can this be merged or are the further comments pending?

File: docs/source/machconf.rst [hunk 2]
```diff
@@ -188,13 +191,23 @@ All fields are mandatory, unless indicated otherwise below.
...
+* ``task_launch_method``: type of compute node access, required for non-MPI tasks. Valid values are: ``SSH``, ``APRUN`` or ``LOCAL``.
 * ``mpi_launch_method``: type of MPI support, required for MPI tasks. Valid values are: ``MPIRUN``, ``MPIEXEC``, ``APRUN``, ``IBRUN``, etc.
 * ``python_interpreter``: path to python (optional).
 * ``python_dist``: `anaconda` or `default`, i.e., not `anaconda` (mandatory).
+* ``cores_per_node``: number of available cores per compute node (optional).
+* ``gpus_per_node``: number of available gpus per compute node (optional).
 * ``pre_bootstrap_0``: list of commands to execute for initialization of main agent (optional).
 * ``pre_bootstrap_1``: list of commands to execute for initialization of sub-agent (optional).
 * ``forward_tunnel_endpoint``: name of the host which can be used to create ssh tunnels from the compute nodes to the outside world (optional).
+* ``virtenv_mode``: RADICAL-Pilot agent enviroment setup (optional):
```
Comment (eirrgang) [inline docs/source/machconf.rst:203]:
```suggestion
* ``virtenv_dist``: By default, RP will install virtualenv version '1.9' while bootstrapping the pilot.  Setting ``virtenv_dist`` to a different version string will use that respective version.  Setting it to the string ``system`` will use the system supplied ``virtualenv`` command.
* ``virtenv_mode``: RADICAL-Pilot agent enviroment setup (optional):
```

**Review by mturilli — APPROVED** [review]
LGTM.

**Pull request merged**